https://github.com/facebook/hhvm/pull/4877
https://github.com/facebook/hhvm/commit/dae5b4c24c47abf6499e9adf8cb6dda1fd1f8df7

From dae5b4c24c47abf6499e9adf8cb6dda1fd1f8df7 Mon Sep 17 00:00:00 2001
From: Yaacov Akiba Slama <ya@slamail.org>
Date: Mon, 23 Feb 2015 11:36:24 -0800
Subject: [PATCH] Add mcrouter include directory after mbfl one to fix
 compilation error

Summary: Fix the following error:
```
Building C object third-party/libmbfl/mbfl/CMakeFiles/mbfl.dir/mbfilter.c.o
In file included from hhvm/third-party/mcrouter/config.h:11:0,
                 from hhvm/third-party/libmbfl/mbfl/mbfilter.c:85:
hhvm/third-party/mcrouter/mcrouter/mcrouter_config.h:21:18: fatal error: chrono: No such file or directory
 #include <chrono>
                  ^
compilation terminated.
third-party/libmbfl/mbfl/CMakeFiles/mbfl.dir/build.make:54: recipe for target 'third-party/libmbfl/mbfl/CMakeFiles/mbfl.dir/mbfilter.c.o' failed
make[2]: *** [third-party/libmbfl/mbfl/CMakeFiles/mbfl.dir/mbfilter.c.o] Error 1
CMakeFiles/Makefile2:450: recipe for target 'third-party/libmbfl/mbfl/CMakeFiles/mbfl.dir/all' failed
make[1]: *** [third-party/libmbfl/mbfl/CMakeFiles/mbfl.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
```
Closes https://github.com/facebook/hhvm/pull/4877

Reviewed By: @sgolemon

Differential Revision: D1864311
---
 CMake/HPHPSetup.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMake/HPHPSetup.cmake b/CMake/HPHPSetup.cmake
index 64468ce..7bcd752 100644
--- a/CMake/HPHPSetup.cmake
+++ b/CMake/HPHPSetup.cmake
@@ -242,15 +242,15 @@ if (NOT FASTLZ_LIBRARY)
   include_directories("${TP_DIR}/fastlz")
 endif()
 
-if (ENABLE_MCROUTER)
-  include_directories("${TP_DIR}/mcrouter")
-endif()
-
 include_directories("${TP_DIR}/timelib")
 include_directories("${TP_DIR}/libafdt/src")
 include_directories("${TP_DIR}/libmbfl")
 include_directories("${TP_DIR}/libmbfl/mbfl")
 include_directories("${TP_DIR}/libmbfl/filter")
+if (ENABLE_MCROUTER)
+  include_directories("${TP_DIR}/mcrouter")
+endif()
+
 add_definitions(-DNO_LIB_GFLAGS)
 include_directories("${TP_DIR}/folly")
 include_directories("${TP_DIR}/thrift")
